File: dual_force_frontend/lib/models/document_metadata.dart

class DocumentMetadata {
  final String id;
  final String filename;
  final String contentType;
  final int sizeBytes;
  final int totalChunks;
  final DateTime createdAt;

  DocumentMetadata({
    required this.id,
    required this.filename,
    required this.contentType,
    required this.sizeBytes,
    required this.totalChunks,
    required this.createdAt,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'filename': filename,
      'contentType': contentType,
      'sizeBytes': sizeBytes,
      'totalChunks': totalChunks,
      'createdAt': createdAt,
    };
  }

  factory DocumentMetadata.fromMap(Map<String, dynamic> map) {
    return DocumentMetadata(
      id: map['id'],
      filename: map['filename'],
      contentType: map['contentType'],
      sizeBytes: map['sizeBytes'],
      totalChunks: map['totalChunks'],
      createdAt: DateTime.parse(map['createdAt']),
    );
  }
}

File: dualForceBackend/src/main/java/com/dualforce/dualForceBackend/entity/documentEntities/DocumentMetadata.java
package com.dualforce.dualForceBackend.entity.documentEntities;

import com.dualforce.dualForceBackend.entity.ExpertSystem;
import com.dualforce.dualForceBackend.entity.User;
import lombok.Getter;
import lombok.Setter;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Setter
@Getter
@Document(collection = "document_metadata")
public class DocumentMetadata {
    @Id
    private ObjectId id;

    private String filename;
    private String contentType;
    private long sizeBytes;
    private int totalChunks;
    private LocalDateTime createdAt;



    public DocumentMetadata() {
        this.createdAt = LocalDateTime.now();
    }


}
